packages/office-addin-sso/src/middle-tier/app.ts

```diff
@@ -76,19 +74,32 @@ export class App {
     });
 
     this.appInstance.get("/getuserdata", async function (req: any, res: any, next: any) {
-      const graphToken = req.get("access_token");
-      const graphUrlSegment = process.env.GRAPH_URL_SEGMENT || "/me";
-      const graphQueryParamSegment = process.env.QUERY_PARAM_SEGMENT || "";
-
-      const graphData = await getGraphData(graphToken, graphUrlSegment, graphQueryParamSegment);
-
-      // If Microsoft Graph returns an error, such as invalid or expired token,
-      // there will be a code property in the returned object set to a HTTP status (e.g. 401).
-      // Relay it to the client. It will caught in the fail callback of `makeGraphApiCall`.
-      if (graphData.code) {
-        next(createError(graphData.code, "Microsoft Graph error " + JSON.stringify(graphData)));
+      const graphTokenResponse = await getAccessToken(req.get("Authorization"));
+      if (graphTokenResponse.claims || graphTokenResponse.error) {
+        graphTokenResponse.claims
+          ? usageDataObject.reportEvent("CliamsResponse")
+          : usageDataObject.reportError(
+              "AccessTokenError",
+              new Error("Access Token Error: " + graphTokenResponse.error)
+            );
+        res.send(graphTokenResponse);
```

Comment:
Seems like this should set an HTTP 400 or 401 based on which error.

Reply:
Sorry I meant 403 (which is what Graph will return) although probably unlikely since you know the URL will always be correct.

Reply:
This a little bit fallout from the access token fetching being on the client before.  I moved it to the middle-tier, but handling the claims and errors that require client action (alternate bootstrap token retrieval) is still in the client and still based on the response.  I'd have to pick apart the response to create a new one with enough information for the client to do what it needs.

Reply:
sound good.

Reply:
I did a little digging to understand how this compared to you sample and this essentially where you wrap the response in a "500" error and the client pulls apart the interior info in order to retry if needed.  The difference here is I'm not wrapping the error and my client handling is part of the sso package and not in the taskpane project code.

I think it could be organized better here . . . but I will leave that for a future change.